### PR TITLE
Mecha Changes

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Mechas now have armour facings.

Attacking from the front results in much less damage, attacking from the side results in normal damage, and hitting the back results in massive bonus damage. (Right now it's 50% less damage from the front, and 50% extra from the back, but those values are easy to tweak, and can be tweaked per mech)

Also lowered the damage EMPs do to mechas, because it's dumb to have 3 departments cooperate to build this armoured box and have security just hose it down in 30 seconds with the ion rifle.

:cl: Kor
rscadd: Mechas now have armour facings. Attacking the front will deal far less damage, while attacking the back will deal massive bonus damage.
rscadd: EMPs now deal far less damage/drain to mechas.
/:cl:

### CODE REVIEW

File: code/game/mecha/mecha_defense.dm [hunk 1]
```diff
@@ -1,7 +1,14 @@
...
+	var/damage_modifier = facing_modifiers[facing_hit]
+	return damage_modifier
 
-/obj/mecha/proc/take_damage(amount, type="brute")
+/obj/mecha/proc/take_damage(amount, type="brute", adir = 0)
 	if(amount)
 		var/damage = absorbDamage(amount,type)
+		if(adir)
+			var/modifier = get_armour_facing(src.dir,adir)
+			damage = damage/modifier
```
Comment (RemieRichards) [inline code/game/mecha/mecha_defense.dm]: ```damage /= modifier```

**Comment by TechnoAlchemisto** [comment]
thirty seconds of firing is a lot of firing

**Comment by RemieRichards** [comment]
I think 50% is too high, I'd go with 25% (for both cases)

**Comment by KorPhaeron** [comment]
25% was actually the value I tested with, and probably far more sane

**Comment by PKPenguin321** [comment]
how does this interact with diagonals

**Comment by RemieRichards** [comment]
@PKPenguin321 it runtimes from dividing by null

**Comment by PKPenguin321** [comment]
fugg

File: code/game/mecha/mecha_defense.dm (hunk 1 above)
Comment (KazeEspada) [inline code/game/mecha/mecha_defense.dm]: use multiplication, easier to manipulate, less chance of a divide by zero error.

**Comment by Lumipharon** [comment]
If you hose it down with the ion, even with this change, you'll most likely just cuck their battery and leave it an intact, but useless tomb.

**Comment by KorPhaeron** [comment]
@Lumipharon 

Wow it takes away half the mechs battery per shot. That seems a bit excessive.

**Comment by Bawhoppen** [comment]
@KorPhaeron well the whole purpose of the ion-rifle is to be anti-armor, so it should be effective

**Comment by KorPhaeron** [comment]
@Bawhoppen 

Difference between "effective" and "my roundstart item effortlessly destroyed the thing that 3+ people collaborated on building over half an hour"

Rock paper scissors is a bad way to balance things.

**Comment by Shapsy** [comment]
Just deploying the ion rifle is a challenge sometimes, I've seen mechs manage to wipe out entire sec teams because the guy with the rifle got tased by a mech and the mech stood on top of it and attacked anyone who tried to grab it

**Comment by xxalpha** [comment]
Don't make it take extra damage from the back, mechas aren't exactly agile things to keep their back from their attackers. This is a huge fucking nerf.

**Comment by GunHog** [comment]
I know how Kor loves to nerf things and what not, especially Science related things, but this is actually interesting to me. The values may be too extreme, though. Directional strength/weakness reminds me of playing Hawken as a heavy mech in turret mode. I can sit in place and tank enemy fire from the front, but I am turtle-slow; If I am hit from the back, I am toast!

I am willing to give this one a shot, if Kor will watch mech combat results and adjusts things as needed.

**Comment by KorPhaeron** [comment]
I don't think shutting it down is very exciting either. Maybe if it stunned the mech instead of permanently fucking it (it is seriously half the cell gone in a single shot).

**Comment by Durkel** [comment]
500 years into the future and emp shielding isn't a thing anymore. Rip mechs.

**Comment by KorPhaeron** [comment]
@xxalpha 

I think having to keep your back protected, with a big reward for doing so (very little damage from the front) is a better weakness and more interesting gameplay than "hope the other guy doesnt have EMPs"

**Comment by xxalpha** [comment]
Then give them EMP protection. This will just make it even easier to tear down a mech with a toolbox.

File: code/game/mecha/mecha_defense.dm [hunk 2]
```diff
@@ -124,6 +131,13 @@
 			deflection *= B.deflect_coeff
 			dam_coeff *= B.damage_coeff
 			break
+/*
+
+	var/facing_modifier = get_armour_facing(src.dir,Proj.dir)
+
+	deflection *= facing_modifier
+	dam_coeff = dam_coeff/facing_modifier
```
Comment (Cheridan) [inline code/game/mecha/mecha_defense.dm]: this bit still needs cleanup

**Comment by KorPhaeron** [comment]
I think this is ready

**Comment by KorPhaeron** [comment]
@AnturK 

Done

**Pull request merged**